pkg/genericapiserver/genericapiserver.go

```diff
@@ -200,7 +206,8 @@ func (s *GenericAPIServer) Run() {
 				// Can't use SSLv3 because of POODLE and BEAST
 				// Can't use TLSv1.0 because of POODLE and BEAST using CBC cipher
 				// Can't use TLSv1.1 because of RC4 cipher usage
-				MinVersion: tls.VersionTLS12,
+				MinVersion:        tls.VersionTLS12,
+				NameToCertificate: namedCerts,
```

Comment:
when I did this in OpenShift, I had to use a custom `GetCertificate` function instead of `NameToCertificate`:

> // Set SNI certificate func
> // Do not use NameToCertificate, since that requires certificates be included in the server's tlsConfig.Certificates list,
> // which we do not control when running with http.Server#ListenAndServeTLS
> GetCertificate: cmdutil.GetCertificateFunc(extraCerts),

can you verify this is no longer the case?

edit: looks like this may have been fixed in go1.5